.docker/Dockerfile

```diff
@@ -0,0 +1,291 @@
+# capture environment to modify layer digest
+RUN env > /tmp/env.txt
+
+################################################################################
+# MARK: seeder - seed workspace artifacts for caching
+################################################################################
+FROM baser AS seeder
+ARG WS_CACHE_ID
+
+ARG CLEAR_WS_CACHE
+RUN --mount=type=cache,id=$WS_CACHE_ID,sharing=private,target=$OVERLAY_WS \
+    if [ -n "$CLEAR_WS_CACHE" ]; then \
+      echo "Clearing cache!" && \
+      rm -rf $OVERLAY_WS/* && \
+      echo "Cache cleared!"; \
+    fi
+
+ARG SEED_WS_CACHE
+RUN --mount=type=bind,source=./,target=/seeder/\
+    --mount=type=cache,id=$WS_CACHE_ID,sharing=private,target=$OVERLAY_WS \
+    if [ -n "$SEED_WS_CACHE" ]; then \
+      echo "Seeding cache!" && \
+      cp -rT /seeder/cache/$OVERLAY_WS $OVERLAY_WS && \
+      echo "Cache seeded!"; \
+    fi
+
+RUN --mount=type=cache,from=cacher,target=/cacher \
+  --mount=type=cache,id=$WS_CACHE_ID,sharing=private,target=$OVERLAY_WS \
+  rm -rf ./src && \
+  cp -r /cacher/$OVERLAY_WS/src ./ && \
+  echo $(date) > /tmp/seeder_stamp.txt
+
+################################################################################
+# MARK: builder - build workspace artifacts for deployment
+################################################################################
+FROM tooler AS builder
+ARG WS_CACHE_ID
+
+# build overlay source
+ARG BUST_BUILD_CACHE
+RUN --mount=type=cache,from=seeder,target=/seeder \
+    --mount=type=cache,id=$WS_CACHE_ID,sharing=private,target=$OVERLAY_WS \
+    . /opt/ros/$ROS_DISTRO/setup.sh && \
+    colcon cache lock \
+      --dirhash-reset && \
+    colcon clean packages -y \
+      --packages-select-cache-invalid \
+      --packages-select-cache-key build \
+      --base-select install && \
+    colcon build \
+      --packages-skip-cache-valid \
+      --mixin $OVERLAY_MIXINS \
+      || true && \
+    echo $(date) > /tmp/builder_stamp.txt
+
+ARG FAIL_ON_BUILD_FAILURE=1
+RUN --mount=type=cache,id=$WS_CACHE_ID,sharing=private,target=$OVERLAY_WS \
+    . /opt/ros/$ROS_DISTRO/setup.sh && \
+    BUILD_FAILED=$( \
+      colcon list \
+        --packages-select-build-failed) && \
+    if [ -n "$BUILD_FAILED" ]; then \
+      echo "BUILD_FAILED: \n$BUILD_FAILED" && \
+      ([ -z "$FAIL_ON_BUILD_FAILURE" ] || exit 1); \
+    fi
+
+################################################################################
+# MARK: dever - setup user account for development
+################################################################################
+FROM $DEV_FROM_STAGE AS dever
+
+# install packages for development work
+COPY .docker/dever_apt_debs.txt /tmp/dever_apt_debs.txt
+RUN --mount=type=cache,sharing=locked,target=/var/cache/apt \
+  cut -d# -f1 < /tmp/dever_apt_debs.txt | envsubst \
+    | xargs apt-get install -y --no-install-recommends
+
+# add default user for devcontainer
+ENV DEV_USER=ubuntu
+RUN echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
```

Comment:
This is a bit dangerous, no?

Reply:
It's for a stage that is used for the dev containers, and just enabled devcontainer tools to remap the ubuntu user's UID/GID to that of the local user on the host, while still facilitating root level access inside the container. The default user for most docker images is (and has always been) root anyway by the way...